crt_portal/static/js/resources.js

```diff
@@ -91,21 +118,154 @@
     }, 2000);
   }
 
-  const selectAllCheckboxes = dom.getElementsByClassName('checkbox-input-all');
-  for (let index = 0; index < selectAllCheckboxes.length; index++) {
-    const parentTable = selectAllCheckboxes[index].closest('.usa-table.crt-table');
-    const allCheckboxes = parentTable.querySelectorAll('td input.usa-checkbox__input');
-    allCheckboxes.forEach(checkbox => {
-      addCheckboxListener(checkbox, index, parentTable);
+  function setUpEventListeners() {
+    const selectAllCheckboxes = dom.getElementsByClassName('checkbox-input-all');
+    for (let index = 0; index < selectAllCheckboxes.length; index++) {
+      const parentTable = selectAllCheckboxes[index].closest('.usa-table.crt-table');
+      const allCheckboxes = parentTable.querySelectorAll('td input.usa-checkbox__input');
+      allCheckboxes.forEach(checkbox => {
+        addCheckboxListener(checkbox, index, parentTable, selectAllCheckboxes);
+      });
+      addCheckAllListener(selectAllCheckboxes[index], allCheckboxes);
+    }
+
+    const copyButtons = document.querySelectorAll('.copy-resource');
+    copyButtons.forEach(function(btn) {
+      btn.onclick = function(event) {
+        event.preventDefault();
+        copyRow(event);
+      };
     });
-    addCheckAllListener(selectAllCheckboxes[index], allCheckboxes);
-  }
 
-  const copyButtons = document.querySelectorAll('.copy-resource');
-  copyButtons.forEach(function(btn) {
-    btn.onclick = function(event) {
+    const searchInput = document.querySelector('#id_search');
+    searchInput.oninput = function(event) {
       event.preventDefault();
-      copyRow(event);
+      setTimeout(() => {
+        makeQuery();
+      }, 1500);
     };
-  });
+
+    const tagInput = document.querySelector('#id_tags-assign-tag');
+    tagInput.onchange = function() {
+      makeQuery();
+    };
```

Comment:
you may want to cancel previous setTimeouts so things don't come back out of order.

https://stackoverflow.com/questions/452003/cancel-kill-window-settimeout-before-it-happens

Reply:
@kewitham I sort of expected `clearTimeout` to be called - am I missing something?

Reply:
Yep I added clearTimeout for every time setTimeout is used. There is no timeout for the tag input event listener.

Reply:
ah mb - must have been looking at the outdated version - thanks for putting up with me :D